src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMModelMethods.scala

```diff
@@ -63,4 +106,13 @@ trait LightGBMModelMethods extends LightGBMModelParams {
   protected def featuresShap(features: Vector): Vector = {
     Vectors.dense(getLightGBMBooster.featuresShap(features))
   }
+
+  protected def updateBoosterParamsBeforePredict(): Unit = {
+    if (getNumIterations != LightGBMConstants.DefaultNumIterations) {
```

Comment:
i dont think these ifs are saving much. COnsider simplifying to just the setters